core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -111,7 +111,7 @@ public QueryValidationException validate() {
     }
 
     @Override
-    public String getName() {
+    public String getWriteableName() {
```

Comment:
I think we don't have this case for queries, cause when we subclass the query usually the base class does everything around serialization. If we did have the case though, like probably aggs do, I wouldn't make this method final and yes we would have to work around the readFrom problem as Colin suggested. Don't see how we can work around it otherwise.